javav2/example_code/s3/src/main/java/com/example/s3/async/PutObjectFromStreamAsync.java

```diff
@@ -0,0 +1,69 @@
+package com.example.s3.async;
+
+// snippet-start:[s3.java2.async_stream.complete]
+// snippet-start:[s3.java2.async_stream.import]
+import com.example.s3.util.AsyncExampleUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.async.AsyncRequestBody;
+import software.amazon.awssdk.core.async.BlockingInputStreamAsyncRequestBody;
+import software.amazon.awssdk.core.exception.SdkException;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.model.PutObjectResponse;
+
+import java.io.ByteArrayInputStream;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+// snippet-end:[s3.java2.async_stream.import]
+
+public class PutObjectFromStreamAsync {
+    private static final Logger logger = LoggerFactory.getLogger(PutObjectFromStreamAsync.class);
+
+
+    public static void main(String[] args) {
+        String bucketName = "x-" + UUID.randomUUID();
+        String key = UUID.randomUUID().toString();
+
+        AsyncExampleUtils.createBucket(bucketName);
+        try {
+            PutObjectFromStreamAsync example = new PutObjectFromStreamAsync();
+            PutObjectResponse putObjectResponse = example.putObjectFromStream(AsyncExampleUtils.client, bucketName, key);
+            logger.info("Object {} etag: {}", key, putObjectResponse.eTag());
+            logger.info("Object {} uploaded to bucket {}.", key, bucketName);
+        } catch (SdkException e) {
+            logger.error(e.getMessage(), e);
+        } finally {
+            AsyncExampleUtils.deleteObject(bucketName, key);
+            AsyncExampleUtils.deleteBucket(bucketName);
+        }
+    }
+
+// snippet-start:[s3.java2.async_stream.main]
+    /**
+     * @param s33CrtAsyncClient - To upload content from a stream of unknown size, <a href="https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/crt-based-s3-client.htmluse">the AWS CRT-based S3 client</a>.
```

Comment:
If this will remain a code comment, add the link in plain text, like so: 
To upload content from a stream of unknown size, use the AWS CRT-based S3 client. For more information, see https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/crt-based-s3-client.html.

Reply:
Content updated per suggestion.